Fix for php7

thomasvargiu opened the pull request:
Fixed indirect variable access for php7 compatibility

marcelog commented:
Hello,

Thanks for your contribution!

This can't be merged, since there are multiple changes in this pull request that are completely unrelated. Also, I do agree about the change for php7, but not the others. Could you please open a separate pull request for that change?

Thanks again! Will close this one now.

thomasvargiu commented:
Why I added these changes?

- Removed ```composer.lock```.
```composer.lock``` should not be included in libraries and it's not considered when you include it in a project. Developing should done with the last dependecies allowed in ```composer.json```.

- Added phpunit in ```require-dev```.
```require-dev``` section in composer is for developing dependecies and allows to point to specific versions of phpunit. Then we could add the project in travis-ci.

- Removed socket timeout test.
This doesn't work, not in php7 at least.
```stream_socket_client``` returns an error immediately. We even should check if the returned value is ```null``` or ```false``` (or not a resource) because in this case it returns ```null``` to me.
It's a unit test, so we should test our methods, not system functions. The only thing you should test is if call the system function with the right parameters.

- Added ```bootstrap``` property on ```phpunit.xml```
```bootstrap.php``` is the default bootstrap script for tests. They doesn't work without it.
We should rename ```phpunit.xml``` in ```phpunit.xml.dist```, ignoring ```phpunit.xml``` in version control.
PHPUnit checks ```phpunit.xml``` first, then ```phpunit.xml.dist``` by default. It allows developers' custom settings if necessary.

I can't add a php7 fix if tests doesn't work.

I would re-open this PR with the changes mentioned above.
Let me know.

marcelog commented:
About 1: Agreed, but in a library it is ignored, so no harm done and this is not needed for this fix. 
About 2: Not needed. If you look at the doc, currently (will be changed soon) everything is done via phing, and the dependencies are installed via pear (again, will be changed soon to be in sync with the latest changes to pagi 2.0.0).
About 3: The test actually is there to see if the connect_timeout option is honored. Will try it on php7 (and soon Travis will do that for us too), but currently it works for me :\
About 4: mm I don't think I agree on this one. Tests should be run in the same way for developers and CI servers (except for configuration -like DBs- which is not the case here).

It's odd that tests don't work for you, they do for me. I'll merge your php7 fix once I leverage PAMI with the changes done for PAGI and Travis will let us know how that worked. :)

Best!
